src/configupdater/parser.py

```diff
@@ -142,14 +161,18 @@ def __init__(
             space_around_delimiters (bool): add a space before and after the
                 delimiter, default True
         """
+        self._document: Document  # bind later
+        self._optionxform_fn = optionxform
+        self._lineno = -1
+        self._fpname = "<???>"
+
         self._filename: Optional[str] = None
         self._space_around_delimiters: bool = space_around_delimiters
 
-        self._dict = OrderedDict  # no reason to let the user change this
+        self._dict = dict  # no reason to let the user change this
```

Comment:
BTW, this change was introduced in the Python's code base, apparently the core team decided to make the standard dictionary the default, instead of the ordered dict, since it now preserves order.
This is a link for the [commit](https://github.com/python/cpython/commit/3a5b0d8988491d9408b22bceea6fd70b91345724) doing the change.

Reply:
Ah, okay, if they do it then let's copy this. Still, since by most definitions a dictionary as a data structure is unordered, being more explicit would be better. Nevertheless, at least if in years the implementation of a dictionary changes again, we are not the only ones with a problem ;-)